client/cmd/job/change_namespace.go

```diff
@@ -118,25 +120,252 @@ func (c *changeNamespaceCommand) sendChangeNamespaceRequest(jobName string) erro
 	return err
 }
 
+func getJobParentsFilePathList(fileFS afero.Fs, jobDirPath string) ([]string, error) {
+	var parentsFilePathList []string
+	var errorMsg string
+	splitDirPaths := strings.Split(jobDirPath, "/")
+	for i := range splitDirPaths {
+		pathNearSpecIdx := len(splitDirPaths) - i
+		parentDirPath := filepath.Join(splitDirPaths[:pathNearSpecIdx]...)
+		parenFilePath := filepath.Join(parentDirPath, "this.yaml")
+		exist, err := afero.Exists(fileFS, parenFilePath)
+		if err != nil {
+			errorMsg += fmt.Sprintf("Unable to read directory %s to check 'this.yaml', err: %s\n", parenFilePath, err.Error())
+			continue
+		}
+		if exist {
+			parentsFilePathList = append(parentsFilePathList, parenFilePath)
+		}
+	}
+	var err error
+	if len(errorMsg) > 0 {
+		err = errors.New(errorMsg)
+	}
+	return parentsFilePathList, err
+}
+
+func mergeYAMLFiles(fs afero.Fs, sourcePath, targetPath string) error {
```

Comment:
Do we need this merge mechanism? Asking it after looking at the implementation of the merge itself, as well as how it will be used by the user.